base/applications/osk/settings.c

```diff
@@ -90,6 +97,34 @@ BOOL LoadDataFromRegistry()
 
     /* Load the sound on click value event */
     Globals.bSoundClick = (dwSoundOnClick != 0);
+
+    /* Query the key */
+    lResult = RegQueryValueExW(hKey,
+                               L"DialogPlacement",
+                               NULL,
+                               &dwType,
+                               (LPBYTE)Buff,
+                               &dwSize);
+
+    if (lResult != ERROR_SUCCESS || dwType != REG_SZ)
+    {
+        /* Bail out and return FALSE if we fail */
+        RegCloseKey(hKey);
+        return FALSE;
+    }
+
+    /* 
+        Parse the data into the buffer and check if such data is valid.
+        Otherwise the dialog will be placed at the bottom of the screen.
+     */
+    if (swscanf(Buff, L"%d %d",
+        &wp.rcNormalPosition.left,
+        &wp.rcNormalPosition.top) == 2)
+    {
+        /* If we're done then use the previous placement coordination (X and Y) data */
+        Globals.PosX = wp.rcNormalPosition.left;
+        Globals.PosY = wp.rcNormalPosition.top;
```

Comment:
What if this is completely off-screen?
Think e.g. monitors plugged in/out, or resolution changed